### PR TITLE
New conducting plate function

### LINKED ISSUE #3979: Impreove conducting plane boundary condition
### Description of the feature

Add conducting plane boundary condition

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by lorenzovecchietti**
Close #3979

Add new function to assign conducting plate boundary condition. Deprecated the old one.
Advantages of the new one:
1. less inputs needed
2. coherency with all the other bc function for icepak
3. now supports transient assignment

Auxiliary functions added.

### CODE REVIEW

File: pyaedt/icepak.py [hunk 1]
```diff
@@ -592,6 +595,12 @@ def create_conduting_plate(
             Boundary object when successful or ``None`` when failed.
 
         """
+
+        warnings.warn(
+            "This method is deprecated in 0.7.8. Use the assign_conducting_plate() method.",
```
Comment (PipKat) [inline pyaedt/icepak.py:600]:
```suggestion
            "This method is deprecated in 0.7.8. Use the ``assign_conducting_plate()`` method.",
```

File: pyaedt/icepak.py [hunk 2]
```diff
@@ -5741,3 +5750,320 @@ def _assign_blower(self, props, faces, inlet_face, fan_curve_flow_unit, fan_curv
...
+            Default is ``"0Cel_m2_per_W"``.
+        shell_conduction : bool, optional
+            Indicates whether shell conduction is considered.
+            Default is ``False``.
+        low_side_rad_material : str, optional
+            Material on the low side for radiation. If ``None``, radiation
+            is disabled on the low side. Default is ``None``.
+        high_side_rad_material : str, optional
+            Material on the high side for radiation. If ``None``, radiation
+            is disabled on the high side. Default is ``None``.
```
Comment (PipKat) [inline pyaedt/icepak.py:5809]:
```suggestion
            Object to assign the boundary to. If a string, specify a surface name.
            If an integer, specify a face ID.
        boundary_name : str, optional
            Boundary name. The default is ``None``, in which case a name is generated automatically.
        total_power : str or float or dict, optional
            Power dissipated by the plate. Thee default is ``"0W"``. If a float,
            the default unit is ``"W"``. A transient or temperature-dependent power
            can be assigned with a dictionary. 
        thermal_specification : str, optional
            Type of condition to apply. The default is `"Thickness"``.
            Options are: ``"Conductance"``, ``"Thermal Impedance"``,
            ``"Thermal Resistance"``, and ``"Thickness"``. 
        thickness : str or float, optional
            If ``thermal_specification="Thickness"`, this parameter represents the
            thickness to model with the plate. The default is ``"1mm"``. If a float,
            the default unit is ``"mm"``. 
        solid_material : str, optional
           If ``thermal_specification="Thickness"`, this parameter represents the
           material of the conducting plate. The default is ``"Al-Extruded"``.
        conductance : str or float, optional
             If ``thermal_specification=`"Conductance"``, this parameter represents the
             conductance of the plate. The default is ``"0W_per_Cel"``. If a float, the default
             unit is ``"W_per_Cel"``. 
        thermal_resistance : str or float, optional
            If ``thermal_specification="Thermal Resistance"``, this parameter represents the
            thermal resistance of the plate. The default is ``"0Kel_per_W"``. If a float, the default
            unit is ``"Kel_per_W"``.
        thermal_impedance : str or float, optional
            If ``thermal_specification="Thermal Impedance"``, this parameter represents the
            thermal impedance of the plate. The default is ``"0Cel_m2_per_W"``. If a float, the
            default unit is "``Cel_m2_per_W"``.           
        shell_conduction : bool, optional
            Whether to consider shell conduction. The default is ``False``.
        low_side_rad_material : str, optional
            Material on the low side for radiation. The default is ``None``, in which
            case radiation is disabled on the low side.
        high_side_rad_material : str, optional
            Material on the high side for radiation. The default is ``None``, in which
            case radiation is disabled on the high side.
```

File: pyaedt/icepak.py [hunk 3]
```diff
@@ -5741,3 +5750,320 @@ def _assign_blower(self, props, faces, inlet_face, fan_curve_flow_unit, fan_curv
...
+            Default is ``"Al-Extruded"``.
+        shell_conduction : bool, optional
+            Indicates whether shell conduction is considered.
+            Default is ``False``.
+        low_side_rad_material : str, optional
+            Material on the low side for radiation. If ``None``,
+            radiation is disabled on the low side. Default is ``None``.
+        high_side_rad_material : str, optional
+            Material on the high side for radiation. If ``None``,
+            radiation is disabled on the high side. Default is ``None``.
```
Comment (PipKat) [inline pyaedt/icepak.py:5903]:
```suggestion
        obj_plate : str or int or list
            Object to assign the boundary to. If a string, specify a surface name.
            If an integer, specify a face ID.
        boundary_name : str, optional
            Boundary name. The default is ``None``, in which case a name is generated
            automatically.
        total_power : str or float or dict, optional
            Power dissipated by the plate. The default is ``"0W"``. If a float, the
            default unit is ``"W"``. A transient or temperature-dependent power
            can be assigned with a dictionary. 
        thermal_resistance : str or float, optional
            If ``thermal_specification="Thermal Resistance"``, this parameter
            represents the thermal resistance of the plate. The default is ``"0Kel_per_W"``.
            If a float, the default unit is ``"Kel_per_W"``.
        solid_material : str, optional
             If ``thermal_specification="Thickness"``, this parameter represents the
             material of the conducting plate. The default is ``"Al-Extruded"``.
        shell_conduction : bool, optional
            Whether to consider shell conduction. The default is ``False``.
        low_side_rad_material : str, optional
            Material on the low side for radiation. The default is ``None``,
            in which case radiation is disabled on the low side.
        high_side_rad_material : str, optional
            Material on the high side for radiation. The default is ``None``,
            in which case radiation is disabled on the high side.
```
These parameter definitions here don't match what's in the function definitoni. For example, ``thermal_specification`` does not appear in the method definition, but ``thickness`` does.

File: pyaedt/icepak.py [hunk 4]
```diff
@@ -5741,3 +5750,320 @@ def _assign_blower(self, props, faces, inlet_face, fan_curve_flow_unit, fan_curv
...
+
+    def assign_conducting_plate_with_resistance(self, obj_plate, boundary_name=None, total_power="0W",
+                                                thermal_resistance="0Kel_per_W",
+                                                shell_conduction=False, low_side_rad_material=None,
+                                                high_side_rad_material=None):
+        """
+        Assign thermal boundary conditions with thickness specification to a conducting plate.
+
+        Parameters
+        ----------
```
Comment (PipKat) [inline pyaedt/icepak.py:5929]: Please format all subsequent "Parameters" sections with the same descriptions from earlier in the PR. Thanks!

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**